src/net_processing.cpp

```diff
@@ -2933,6 +2970,59 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    // Received from a peer demonstrating readiness to announce transactions via reconciliations.
+    // This feature negotiation should happen: between VERSION and VERACK to avoid relay problems
+    // from switching announcement protocols after the connection is up.
+    if (msg_type == NetMsgType::SENDRECON) {
+        // For now, transaction reconciliation is an experimental feature that could be
+        // enabled via CLI.
+        if (!m_txreconciliation) {
+            LogPrint(BCLog::NET, "sendrecon from peer=%d ignored according to our config\n", pfrom.GetId());
+            return;
+        }
+
+        if (pfrom.fSuccessfullyConnected) {
+            // Disconnect peers that send a SENDRECON message after VERACK.
+            LogPrint(BCLog::NET, "sendrecon received after verack from peer=%d; disconnecting\n", pfrom.GetId());
+            pfrom.fDisconnect = true;
+            return;
+        }
+
+        if (!peer->m_tx_relay) {
+            // Disconnect peers that send a SENDRECON message even though we indicated we don't
+            // support transaction relay.
+            LogPrint(BCLog::NET, "sendrecon received from peer=%d to which we indicated no tx relay; disconnecting\n", pfrom.GetId());
+            pfrom.fDisconnect = true;
```

Comment:
Mutating this value doesn't trigger a functional test failure.
If you can add one more test to check that disconnection is enforced in case we receive a SENDRECON even if we signaled we don't support transaction relay.